### ISSUE TITLE
Generate Observable instead of Promise

### ISSUE THREAD

**Issue by Nivelium** [protobufjs/protobuf.js#1355]
protobuf.js version: <6.8.8>

I'm using Nest.js for development and gRPC for microservices communication. And it's quite uncomfortable injecting a client. Because nest.js client returns ```Observable```.  So i would like for pbjs to have a flag smth like ``--observable-instead-of-promise```.

When i have e.g. two grpc methods:
```
service PersonService {
    rpc GetPersons(GetPersonsRequest) returns (GetPersonsResult);
    rpc GetOnePerson(GetOnePersonRequest) returns (GetOnePersonResult);
}
```
I have generated types for this something like:
```
class PersonService extends $protobuf.rpc.Service {
        constructor(rpcImpl: $protobuf.RPCImpl, requestDelimited?: boolean, responseDelimited?: boolean);
        public static create(rpcImpl: $protobuf.RPCImpl, requestDelimited?: boolean, responseDelimited?: boolean): PersonService;

        public getPersons(request: genom.IGetPersonsRequest, callback: genom.PersonService.GetPersonsCallback): void;

        public getPersons(request: genom.IGetPersonsRequest): **Promise**<genom.GetPersonsResult>;

        public getOnePerson(request: genom.IGetOnePersonRequest, callback: genom.PersonService.GetOnePersonCallback): void;

        public getOnePerson(request: genom.IGetOnePersonRequest): **Promise**<genom.GetOnePersonResult>;
    }
```

The return type of methods is ```**Promise**<Something>```

So i have to do something like in my service file instead of just importing gRPC service and using it as class.
```
abstract class IPersonService {
    abstract getPersons(data: IGetPersonsRequest): Observable<IGetPersonsResult>
    abstract getOnePerson(data: IGetOnePersonRequest): Observable<GetOnePersonResult>
}
```

**Comment by pereslavtsev** [comment]
I have the same problem. For myself I created a temporary solution using a library [replace-in-file](https://www.npmjs.com/package/replace-in-file). After every type generation, I run the shell script:

```shell
OUT_DIR="${PWD}/compiled" # your ts-files

replace-in-file \
  /Promise/g \
  Observable \
  "${OUT_DIR}/compiled.d.ts" \
  --isRegex

replace-in-file \
  /\"protobufjs\"\;/g \
  "\"protobufjs\"; import { Observable } from \"rxjs\";" \
  "${OUT_DIR}/compiled.d.ts" \
  --isRegex
```

This looks bad but it solves the problem with types in nest.js

**Comment by drymek** [comment]
probably `sed` is more popular tool to replace:

```bash
sed -i 's/Promise/Observable/g' types.d.ts && sed -i '1i import { Observable } from "rxjs";' types.d.ts
```

Worth to notice, there is another solution: https://github.com/AlexDaSoul/nestjs-proto-gen-ts
They parse file with https://github.com/AlexDaSoul/nestjs-proto-gen-ts/blob/bcfb60377ff4ef526ce1bfaaa14fe8c3a3b1f2fd/src/compiller/index.ts#L97 and render output with handlebars: 
https://github.com/AlexDaSoul/nestjs-proto-gen-ts/blob/00d7975bba4d365eca4db7160fc36d40b5729275/templates/nestjs-grpc.hbs#L44

**Comment by ericmorand** [comment]
This issue does not belong here, in my opinion. Nest.js should provide a way to use Promise instead of Observable. If protobuf.js starts to implement things based on Nest.js expectations, why not doing the same for every other library that depends on protobuf.js?

**Comment by ssilve1989** [comment]
> Nest.js should provide a way to use Promise instead of Observable

@ericmorand I disagree. Especially when dealing with ServerStream, ClientStream, BidiStream requests. Observable is the most concise way to handle and represent those contracts. A promise doesn't really even make sense for an infinite stream or a finite stream with a time dimension. 

In the past I have used the JSON descriptors to generate my own service interfaces that has the appropriate signatures but it would be great if protobufjs offered an option to generate service interfaces that work for Nest like other protogen libraries do (ts-proto for example) and seeing as how Nest is rising in popularity extremely fast for handling gRPC services.

**Comment by ericmorand** [comment]
Still, Observable is a library-specific thing provided by RxJS. It is an implementation among others of event emitter. There is not reason to support this implementation and not another one.

And Nest may be rising in popularity but it does not mean that it is the best choice for micro-services. Once again, it is not the responsibility of protobuf.js to decide what library is popular in place of the users of protobuf.js.

**Comment by dcodeIO** [comment]
Closing this issue because it has been inactive for more than two years and the project has changed substantially.

If this is still relevant, please open a fresh issue with current reproduction steps, version details, and any updated context.

Please also consider that maintaining protobuf.js takes time and effort. If the project is valuable to you or your organization, sponsorship is appreciated.

Thanks!
